dali/base/dadfs.cpp

```diff
@@ -7549,6 +7550,17 @@ bool CDistributedFileDirectory::isSuperFile(    const char *logicalname,
     return tree.get()&&(strcmp(tree->queryName(),queryDfsXmlBranchName(DXB_SuperFile))==0);
 }
 
+bool CDistributedFileDirectory::isSuperIndexFile(const char *logicalname,
+    IUserDescriptor *user, INode *foreigndali, unsigned timeout)
+{
+    Owned<IPropertyTree> tree = getFileTree(logicalname, user, foreigndali, timeout, false);
+    if (!tree.get())
```

Comment:
I am concerned with [lack of] efficiency of this call and the previous isSuperFile(),
especially as the calling code may call both..

If it's a foreign file - you may have to do a getFileTree(), but should avoid doing it twice.

I would not introduce this new method and not use isSuperFile(), but instead
call something that returns you a IFileDescriptor or a IDistributedFile (e.g. IDistributedFileDirectory::getFileDescriptor()) - and test that for super and index.

Reply:
@jakesmith I think the current implementation is only the first half of the question. The second part is (I think): Is this a duplicated solution of an already merged PR?
And the bonus (design/behaviour/educational) question is  Should we hide a user/ECL programmer error when they wants to copy a superkey file as a simple logical file.

Reply:
From a IDistributedFile you should be able to tell:
1) if it's a super (cheaply) by querying querySuperFile().
2) if it's a index/key file (cheaply) via isFileKey()

So afaik, you should be able to do a normal lookup (that returns you a IDistributedFile) , then test.
And as per previous comment - throw an exception if the're attempting to copy a super index without preserving the superfile.

Reply:
The DFU server (in the #12606) already throws an exception. 
Hidden in the ESP this code to fix the programmer/user error and my question is: do we really want to do this?

Reply:
See my other comments.
https://github.com/hpcc-systems/HPCC-Platform/pull/12674#discussion_r295375113